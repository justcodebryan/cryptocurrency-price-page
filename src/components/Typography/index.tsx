import { DetailedHTMLProps, HTMLAttributes, PropsWithChildren } from 'react'

type TypographyProps = PropsWithChildren<DetailedHTMLProps<HTMLAttributes<HTMLDivElement>, HTMLDivElement>>

const Typography = ({ children, ...rest }: TypographyProps) => {
  return <div {...rest}>{children}</div>
}

export default Typography
