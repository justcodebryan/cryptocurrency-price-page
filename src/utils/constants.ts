import { NoopFn } from '@/types'

export const NOOP: NoopFn = () => void 0

export const DEFAULT_MODAL_CONTAINER_ID = 'modal-container'

export const DEFAULT_POLLING_INTERVAL = 10 * 1000
