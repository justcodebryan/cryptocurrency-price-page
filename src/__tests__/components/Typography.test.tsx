import Typography from '@/components/Typography'
import { toJSON } from '@/utils/test-utils'
import renderer from 'react-test-renderer'
import { render, screen } from '@testing-library/react'

describe('Typography Test Suit', () => {
  test('Typography Snapshot', () => {
    const component = renderer.create(<Typography />)
    const tree = toJSON(component)
    expect(tree).toMatchSnapshot()
  })

  test('Should show children element', () => {
    const { container } = render(<Typography>test</Typography>)
    expect(container.querySelector('div')).toBeDefined()
    expect(screen.getByText('test')).toBeDefined()
  })
})
