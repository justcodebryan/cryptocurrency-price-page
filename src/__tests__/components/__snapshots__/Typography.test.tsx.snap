// Vitest Snapshot v1

exports[`Typography Test Suit > Typography Snapshot 1`] = `<div />`;
